opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -161,6 +161,14 @@ impl HttpExporterBuilder {
                     .unwrap(), // TODO: Return ExporterBuildError::ThreadSpawnFailed
                 ) as Arc<dyn HttpClient>);
             }
+            #[cfg(any(
+                all(feature = "hyper-client", feature = "reqwest-client"),
+                all(feature = "hyper-client", feature = "reqwest-blocking-client"),
+                all(feature = "reqwest-client", feature = "reqwest-blocking-client")
+            ))]
+            {
+                panic!("Can't enable more than one HTTP client features simultaneously. Please choose only one: hyper-client, reqwest-client, or reqwest-blocking-client (default feature)");
```

Comment:
could we return a Err here (https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-otlp/src/exporter/mod.rs#L107-L137) instead of panic?

Reply:
I changed it to a normal error. `compile_error!` would work, but it would also fail clippy linting with all features enabled 😅